kore/src/Kore/Step/Simplification/InjSimplifier.hs

```diff
@@ -66,11 +80,12 @@ data InjSimplifier = InjSimplifier
       --
       --        Returns 'Distinct' if the sort injections cannot match, or 'Unknown' if
       --        further simplification could produce matching injections.
-      unifyInj ::
+      unifyInjs ::
         forall variable.
         InternalVariable variable =>
         Inj (TermLike variable) ->
         Inj (TermLike variable) ->
+        Either Distinct InjUnify ->
```

Comment:
`InjUnify` should contain all the data which `unifyInjs` needs to do its job. This will minimize the risk that `unifyInjs` is called with the wrong `Inj`s for the supplied match data.